### ISSUE TITLE
Integrate Hermite moment RHS into timestepping

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#49]
## Context
Issue #22 (PR #46) implemented the Hermite moment RHS functions but didn't integrate them into the timestepper. Currently `timestepping.py:136` returns zeros for `dg_dt`:

```python
# Hermite moment evolution (Issues #22-24 - not yet implemented)
dg_dt = jnp.zeros_like(fields.g)
```

The physics functions are **fully implemented and tested** (11 passing tests):
- `g0_rhs()`: Density perturbation evolution (thesis Eq. 2.7)
- `g1_rhs()`: Parallel velocity evolution (thesis Eq. 2.8)  
- `gm_rhs()`: Higher moment cascade (thesis Eq. 2.9)

But they're not called from the timestepper!

## Goal
Hook up the existing Hermite RHS functions into `_krmhd_rhs_jit()` to enable full kinetic KRMHD simulations.

### 1. Update `_krmhd_rhs_jit()` signature
Add required parameters:
- `beta_i: float`
- `nu: float` (collision frequency)
- `Lambda: float` (for g1_rhs)
- `M: int` (maximum moment index)

### 2. Replace zero placeholder with actual RHS computation
```python
# In _krmhd_rhs_jit():
# Replace: dg_dt = jnp.zeros_like(fields.g)

# With loop over moments:
dg_dt_0 = g0_rhs(fields.g, fields.z_plus, fields.z_minus, kx, ky, kz, 
                 dealias_mask, beta_i, Nz, Ny, Nx)
dg_dt_1 = g1_rhs(fields.g, fields.z_plus, fields.z_minus, kx, ky, kz,
                 dealias_mask, beta_i, Lambda, Nz, Ny, Nx)

# Loop for m >= 2
dg_dt = fields.g.at[:, :, :, 0].set(dg_dt_0)
dg_dt = dg_dt.at[:, :, :, 1].set(dg_dt_1)
for m in range(2, M+1):
    dg_dt_m = gm_rhs(fields.g, fields.z_plus, fields.z_minus, kx, ky, kz,
                     dealias_mask, m, beta_i, nu, Nz, Ny, Nx)
    dg_dt = dg_dt.at[:, :, :, m].set(dg_dt_m)
```

### 3. Update function signatures to pass parameters
- `krmhd_rhs()`: Extract `beta_i`, `nu`, `Lambda`, `M` from state
- `gandalf_step()`: Thread parameters through to RHS function

### 4. Add integration test
Create `test_full_krmhd_evolution()` that:
- Initializes both Elsasser fields AND Hermite moments
- Runs several timesteps
- Verifies g moments evolve (not zero)
- Checks energy conservation includes kinetic contribution

## Testing
- [ ] All existing 80 tests still pass
- [ ] New integration test with non-zero g evolution
- [ ] Verify kinetic energy contribution calculated correctly
- [ ] Test with M=0 (fluid limit) matches Elsasser-only runs

## Dependencies
- Issue #22 (completed - provides RHS functions)
- PR #46 (merged - RHS implementation)
- PR #47 (merged - GANDALF timestepper)

## Next Steps After This
- Issue #23: Enhanced collision operators
- Issue #24: Hermite closures
- Issue #10: Linear physics validation (now with kinetic effects!)

## Labels
enhancement, kinetic-physics
